### PR TITLE
Modify partition's version name to what it means

### PR DESCRIPTION
**Pull request by morningman**
1. committedVersion(Hash) -> visibleVersion(Hash)
2. currentVersion(Hash) -> committedVersion(Hash)
3. add some comment to make the code more readable.
4. fix a ut bug that editlog may be null in CatalogIdGenerator.

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/transaction/PublishVersionDaemon.java [hunk 1]
```diff
@@ -68,8 +68,8 @@ private void publishVersion() {
         // attention here, we publish transaction state to all backends including dead backend, if not publish to dead backend
         // then transaction manager will treat it as success
         List<Long> allBackends = Catalog.getCurrentSystemInfo().getBackendIds(false);
-        if (allBackends == null || allBackends.size() == 0) {
-            LOG.warn("some transaction state need to publish, but no alive backends!!!");
+        if (allBackends.isEmpty()) {
+            LOG.warn("some transaction state need to publish, but no backend exists");
             return;
```
Comment (yiguolei) [inline fe/src/main/java/org/apache/doris/transaction/PublishVersionDaemon.java]: if allbackends == null ,then nptr throws

File: fe/src/main/java/org/apache/doris/transaction/PublishVersionDaemon.java [hunk 2]
```diff
@@ -122,15 +122,20 @@ private void publishVersion() {
         
         TabletInvertedIndex tabletInvertedIndex = Catalog.getCurrentInvertedIndex();
         // try to finish the transaction, if failed just retry in next loop
+        long currentTime = System.currentTimeMillis();
         for (TransactionState transactionState : readyTransactionStates) {
+            if (transactionState.getPublishVersionTime() - currentTime < Config.publish_version_interval_millis * 2) {
+                // wait 2 rounds before handling publish result
```
Comment (yiguolei) [inline fe/src/main/java/org/apache/doris/transaction/PublishVersionDaemon.java]: should be currentTime - publishVersionTime

File: fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java [hunk 3]
```diff
@@ -448,35 +454,40 @@ public void abortTransaction(Long dbId, String label, String reason) throws User
                 continue;
             }
             db.readLock();
-            writeLock();
             try {
```
Comment (yiguolei) [inline fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java]: why not use write lock

File: fe/src/main/java/org/apache/doris/transaction/PublishVersionDaemon.java (hunk 1 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/transaction/PublishVersionDaemon.java]: allbackends never returns null

File: fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java (hunk 3 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java]: There is no changing here, so read lock is enough.

**Review by yiguolei — APPROVED** [review]

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**